test/connection_test.py

```diff
@@ -108,7 +108,7 @@ def test_initialize_raises_argument_error_on_missing_site(self):
     def test_get(self):
         person = util.to_json({'id': 1, 'name': 'Matz'}, root='person')
         self.http.respond_to(
-            'GET', 'http://localhost/people/1.json', {}, person)
+            'GET', '/people/1.json', {}, person)
```

Comment:
The new `HTTPSConnectionTest` class mirrors all of the tests in `ConnectionTest`, but with calls to `https://`URLs. Not changing these test would have meant either overriding each individual test that explicitly made a request using a full domain and protocol in the new `HTTPSConnectionTest` class, or altering `ConnectionTest` anyway to build the URLs with something like:

```python
'GET', self.http.site + '/people/1.json', {}, person)
```

As it was only a couple of tests that used explicit URLs and it doesn't affect the behaviour of the `respond_to` method, I figured this was the cleanest solution.